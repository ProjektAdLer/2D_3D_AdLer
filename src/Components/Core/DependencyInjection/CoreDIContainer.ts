import { LoadWorldController } from "../Presentation/React/LoadWorldButton/LoadWorldController";
import ILoadWorldController from "../Presentation/React/LoadWorldButton/ILoadWorldController";
import { Container } from "inversify";
import CORE_TYPES from "./CoreTypes";
import ISceneView from "../Presentation/Babylon/SceneManagement/ISceneView";
import SceneView from "../Presentation/Babylon/SceneManagement/SceneView";
import SceneViewModel from "../Presentation/Babylon/SceneManagement/SceneViewModel";
import ScenePresenter from "../Presentation/Babylon/SceneManagement/ScenePresenter";
import MainScene from "../Presentation/Babylon/SceneManagement/MainScene";
import ICreateSceneClass from "../Presentation/Babylon/SceneManagement/ICreateSceneClass";
import IScenePresenter from "../Presentation/Babylon/SceneManagement/IScenePresenter";
import infrastructureDIContainer from "./InfrastructureDIContainer";
import useCaseDIContainer from "./UseCases/UseCaseDIConatiner";
import PortsDIContainer from "./Ports/PortsDIContainer";
import BuilderDIContainer from "./Builders/BuilderDIContainer";
import INavigation from "../Presentation/Babylon/Navigation/INavigation";
import Navigation from "../Presentation/Babylon/Navigation/Navigation";
import NavigationConfiguration from "../Presentation/Babylon/Navigation/NavigationConfiguration";

var CoreDIContainer = new Container();

// Scene
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind(SceneViewModel).toSelf().inSingletonScope();
CoreDIContainer.bind<IScenePresenter>(CORE_TYPES.IScenePresenter)
  .to(ScenePresenter)
  .inSingletonScope();

// Navigation
CoreDIContainer.bind<INavigation>(CORE_TYPES.INavigation)
  .to(Navigation)
  .inSingletonScope();
CoreDIContainer.bind(NavigationConfiguration).toSelf();

// Controllers
CoreDIContainer.bind<ILoadWorldController>(CORE_TYPES.ILoadWorldController)
  .to(LoadWorldController)
  .inSingletonScope();

// bind other CreateSceneClass here for testing puposes -MK
CoreDIContainer.bind<ICreateSceneClass>(CORE_TYPES.ICreateSceneClass).to(
  MainScene
);

CoreDIContainer.load(infrastructureDIContainer);
CoreDIContainer.load(useCaseDIContainer);
CoreDIContainer.load(PortsDIContainer);
CoreDIContainer.load(BuilderDIContainer);

export default CoreDIContainer;
