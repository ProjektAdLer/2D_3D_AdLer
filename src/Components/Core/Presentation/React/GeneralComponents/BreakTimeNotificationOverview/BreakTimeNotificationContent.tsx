import { useState } from "react";
import { useTranslation } from "react-i18next";
import IBreakTimeNotification from "src/Components/Core/Domain/BreakTimeNotifications/IBreakTimeNotification";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";

export default function BreakTimeNotificationContent({
  breakTimeNotification,
}: {
  breakTimeNotification: IBreakTimeNotification;
}) {
  const [currentSlide, setCurrentSlide] = useState(0);

  const { t: translate } = useTranslation("breakTime");

  return (
    <div className="max-h-[90vh] pb-4 portrait:max-w-[90vw]">
      <div className="slider-wrapper mx-0 my-auto flex flex-col items-center gap-6">
        <div className="slider flex max-h-[90vh] flex-col items-center justify-center gap-4 overflow-y-auto portrait:max-w-[90vw]">
          {/* Description Title Text */}
          <p className="w-full max-w-[60vw] pb-4 pl-6 text-sm font-bold text-adlerdarkblue md:text-xl lg:text-xl ultraWide:max-w-[40vw] mobile-landscape:pb-0 mobile-landscape:text-2xs mobile-portrait:max-w-[90dvw]">
            {breakTimeNotification.titleMessageKeys.map((key) => {
              return translate(key) + " ";
            })}
          </p>

          {/* Image Slider */}
          <div className="align-center flex w-full flex-row justify-center">
            {/* Button left*/}
            <div className="flex w-12 items-center justify-start">
              {currentSlide > 0 && (
                <StyledButton
                  shape="smallSquare"
                  className="w-full rounded-xl bg-whitetrans"
                  onClick={() => {
                    setCurrentSlide(currentSlide - 1);
                  }}
                  title={translate("previousSlideToolTip").toString()}
                  data-testid="breaktime-previousSlide"
                >
                  <p className="rotate-180">&#10140;</p>
                </StyledButton>
              )}
            </div>
            {/* Image Slide */}
            <img
              id="slide-2"
              className="max-h-[60vh] rounded-lg object-cover portrait:max-w-[70vw]"
              src={breakTimeNotification.images[currentSlide]}
              title="Pausenhinweis"
              alt=""
            />
            {/* Button right*/}
            <div className="flex w-12 items-center justify-end">
              {currentSlide < breakTimeNotification.images.length - 1 && (
                <StyledButton
                  shape="smallSquare"
                  className="w-full rounded-xl bg-whitetrans"
                  onClick={() => {
                    setCurrentSlide(currentSlide + 1);
                  }}
                  title={translate("nextSlideToolTip").toString()}
                  data-testid="breaktime-nextSlide"
                >
                  <p className="">&#10140;</p>
                </StyledButton>
              )}
            </div>
          </div>
        </div>

        {/* Image Slide Navigation Dots */}
        {breakTimeNotification.images.length > 1 && (
          <div className="slider-nav z-10 flex gap-4 lg:gap-6">
            {breakTimeNotification.images.map((_, index) => {
              return (
                <button
                  className={
                    (currentSlide === index ? "bg-yellow-400 " : "bg-white ") +
                    "h-2 w-2 rounded-full opacity-75 transition duration-200 ease-in-out hover:opacity-100 lg:h-4 lg:w-4"
                  }
                  data-testid={"breakSliderButton" + index}
                  onClick={() => setCurrentSlide(index)}
                />
              );
            })}
          </div>
        )}
      </div>
    </div>
  );
}
