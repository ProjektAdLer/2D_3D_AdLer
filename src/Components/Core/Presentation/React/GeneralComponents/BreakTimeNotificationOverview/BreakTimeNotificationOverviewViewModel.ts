import IBreakTimeNotification from "src/Components/Core/Domain/BreakTimeNotifications/IBreakTimeNotification";

export default class BreakTimeNotificationOverviewViewModel {
  shortBreakTimeNotifications: IBreakTimeNotification[];
  mediumBreakTimeNotifications: IBreakTimeNotification[];
  longBreakTimeNotifications: IBreakTimeNotification[];
}
