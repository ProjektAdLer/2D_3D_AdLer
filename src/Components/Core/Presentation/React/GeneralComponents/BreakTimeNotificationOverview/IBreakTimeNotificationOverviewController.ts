import IBreakTimeNotification from "src/Components/Core/Domain/BreakTimeNotifications/IBreakTimeNotification";

export default interface IBreakTimeNotificationOverviewController {
  selectNotification(notification: IBreakTimeNotification): void;
  returnToOverview(): void;
  closeModal(): void;
}
