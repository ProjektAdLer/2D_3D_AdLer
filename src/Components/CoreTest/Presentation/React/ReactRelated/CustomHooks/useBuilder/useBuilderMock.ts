import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
