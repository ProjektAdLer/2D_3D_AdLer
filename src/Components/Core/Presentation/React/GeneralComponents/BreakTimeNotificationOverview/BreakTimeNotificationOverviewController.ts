import type IBreakTimeNotification from "src/Components/Core/Domain/BreakTimeNotifications/IBreakTimeNotification";
import IBreakTimeNotificationOverviewController from "./IBreakTimeNotificationOverviewController";
import BreakTimeNotificationOverviewViewModel from "./BreakTimeNotificationOverviewViewModel";
import bind from "bind-decorator";

export default class BreakTimeNotificationOverviewController
  implements IBreakTimeNotificationOverviewController
{
  constructor(private viewModel: BreakTimeNotificationOverviewViewModel) {}

  @bind
  selectNotification(notification: IBreakTimeNotification): void {
    this.viewModel.selectedNotification.Value = notification;
  }

  @bind
  returnToOverview(): void {
    this.viewModel.selectedNotification.Value = null;
  }

  @bind
  closeModal(): void {
    this.viewModel.showModal.Value = false;
    this.viewModel.selectedNotification.Value = null;
  }
}
