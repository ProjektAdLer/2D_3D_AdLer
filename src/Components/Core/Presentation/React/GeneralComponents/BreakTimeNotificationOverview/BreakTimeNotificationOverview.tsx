import IBreakTimeNotificationOverviewController from "./IBreakTimeNotificationOverviewController";
import BreakTimeNotificationOverviewViewModel from "./BreakTimeNotificationOverviewViewModel";
import useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";
import BUILDER_TYPES from "~DependencyInjection/Builders/BUILDER_TYPES";
import StyledModal from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledModal";
import { useTranslation } from "react-i18next";
import useObservable from "~ReactComponents/ReactRelated/CustomHooks/useObservable";
import IBreakTimeNotification from "src/Components/Core/Domain/BreakTimeNotifications/IBreakTimeNotification";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";
import BreakTimeNotificationContent from "./BreakTimeNotificationContent";

function BreakTimeNotificationThumbnail({
  notification,
  onClick,
}: {
  notification: IBreakTimeNotification;
  onClick: () => void;
}) {
  const { t: translate } = useTranslation("breakTime");

  return (
    <div>
      <StyledButton onClick={onClick} data-testid={notification.titleKey}>
        <img
          src={notification.images[0]}
          alt={translate(notification.titleKey) + "Thumbnail"}
        />
      </StyledButton>
      <div onClick={onClick}>{translate(notification.titleKey)}</div>
    </div>
  );
}

function BreakTimeNotificationThumbnailSection({
  title,
  notifications,
  controller,
}: {
  title: string;
  notifications: IBreakTimeNotification[];
  controller: IBreakTimeNotificationOverviewController;
}) {
  return (
    <div>
      <h1>{title}</h1>
      <div className="grid grid-cols-5 gap-5">
        {notifications.map((notification) => (
          <BreakTimeNotificationThumbnail
            key={notification.titleKey}
            notification={notification}
            onClick={() => controller.selectNotification(notification)}
          />
        ))}
      </div>
    </div>
  );
}

export default function BreakTimeNotificationOverview() {
  const [viewModel, controller] = useBuilder<
    BreakTimeNotificationOverviewViewModel,
    IBreakTimeNotificationOverviewController
  >(BUILDER_TYPES.IBreakTimeNotificationOverviewBuilder);

  const [showModal] = useObservable(viewModel?.showModal);
  const [selectedNotification] = useObservable(viewModel?.selectedNotification);

  const { t: translate } = useTranslation("breakTime");

  if (!viewModel || !controller) return null;
  if (!showModal) return null;

  return (
    <StyledModal
      title={translate("overviewTitle")!}
      onClose={controller.closeModal}
      showModal={showModal}
    >
      {/* Selection */}
      {!selectedNotification && (
        <div className="">
          <BreakTimeNotificationThumbnailSection
            title={translate("shortBreaks")}
            notifications={viewModel.shortBreakTimeNotifications}
            controller={controller}
          />
          <BreakTimeNotificationThumbnailSection
            title={translate("mediumBreaks")}
            notifications={viewModel.mediumBreakTimeNotifications}
            controller={controller}
          />
          <BreakTimeNotificationThumbnailSection
            title={translate("longBreaks")}
            notifications={viewModel.longBreakTimeNotifications}
            controller={controller}
          />
        </div>
      )}

      {/* Single Notification */}
      {selectedNotification && (
        <div>
          <BreakTimeNotificationContent
            breakTimeNotification={selectedNotification!}
          />
          <StyledButton onClick={controller.returnToOverview}>
            {"<"}
          </StyledButton>
        </div>
      )}
    </StyledModal>
  );
}
