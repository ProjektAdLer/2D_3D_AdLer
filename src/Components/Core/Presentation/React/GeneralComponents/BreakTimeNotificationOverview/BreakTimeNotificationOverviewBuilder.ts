import { injectable } from "inversify";
import BreakTimeNotificationOverviewController from "./BreakTimeNotificationOverviewController";
import BreakTimeNotificationOverviewPresenter from "./BreakTimeNotificationOverviewPresenter";
import IBreakTimeNotificationOverviewController from "./IBreakTimeNotificationOverviewController";
import IBreakTimeNotificationOverviewPresenter from "./IBreakTimeNotificationOverviewPresenter";
import BreakTimeNotificationOverviewViewModel from "./BreakTimeNotificationOverviewViewModel";
import PresentationBuilder from "../../../PresentationBuilder/PresentationBuilder";

/*
This Template Provides the whole scaffolding for a React Component.
Copy below lines in the DI Container and its Types

bind<IPresentationBuilder>(BUILDER_TYPES.IBreakTimeNotificationOverviewBuilder).to(BreakTimeNotificationOverviewBuilder);
IBreakTimeNotificationOverviewBuilder: Symbol("IBreakTimeNotificationOverviewBuilder"),

director.Builder = CoreDIContainer.get<IPresentationBuilder>(
  BUILDER_TYPES.IBreakTimeNotificationOverviewBuilder
);
director.build();
*/

@injectable()
export default class BreakTimeNotificationOverviewBuilder extends PresentationBuilder<
  BreakTimeNotificationOverviewViewModel,
  IBreakTimeNotificationOverviewController,
  undefined,
  IBreakTimeNotificationOverviewPresenter
> {
  constructor() {
    super(
      BreakTimeNotificationOverviewViewModel,
      BreakTimeNotificationOverviewController,
      undefined,
      BreakTimeNotificationOverviewPresenter,
    );
  }
}
