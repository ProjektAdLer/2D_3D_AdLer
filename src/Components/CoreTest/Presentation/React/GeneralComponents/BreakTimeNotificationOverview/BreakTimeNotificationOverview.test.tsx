import { act, fireEvent, render } from "@testing-library/react";
import React from "react";
import BreakTimeNotificationOverview from "../../../../../Core/Presentation/React/GeneralComponents/BreakTimeNotificationOverview/BreakTimeNotificationOverview";
import { Provider } from "inversify-react";
import CoreDIContainer from "../../../../../Core/DependencyInjection/CoreDIContainer";
import useBuilderMock from "../../ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import BreakTimeNotificationOverviewViewModel from "../../../../../Core/Presentation/React/GeneralComponents/BreakTimeNotificationOverview/BreakTimeNotificationOverviewViewModel";
import "@testing-library/jest-dom";
import IBreakTimeNotificationOverviewController from "../../../../../Core/Presentation/React/GeneralComponents/BreakTimeNotificationOverview/IBreakTimeNotificationOverviewController";
import mock from "jest-mock-extended/lib/Mock";

describe("BreakTimeNotificationOverview", () => {
  test("should render", () => {
    const viewModel = new BreakTimeNotificationOverviewViewModel();
    viewModel.showModal.Value = true;
    useBuilderMock([
      viewModel,
      mock<IBreakTimeNotificationOverviewController>(),
    ]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <BreakTimeNotificationOverview />
      </Provider>,
    );

    expect(renderResult.container).not.toBeEmptyDOMElement();
    expect(renderResult.container).toMatchSnapshot();
  });

  test("should not render without view model", () => {
    useBuilderMock([undefined, undefined]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <BreakTimeNotificationOverview />
      </Provider>,
    );

    expect(renderResult.container.firstChild).toBeNull();
  });

  test("should not render without view model show modal value", () => {
    const viewModel = new BreakTimeNotificationOverviewViewModel();
    viewModel.showModal.Value = false;
    useBuilderMock([
      viewModel,
      mock<IBreakTimeNotificationOverviewController>(),
    ]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <BreakTimeNotificationOverview />
      </Provider>,
    );

    expect(renderResult.container.firstChild).toBeNull();
  });

  test("should not render without controller", () => {
    const viewModel = new BreakTimeNotificationOverviewViewModel();
    viewModel.showModal.Value = true;
    useBuilderMock([viewModel, undefined]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <BreakTimeNotificationOverview />
      </Provider>,
    );

    expect(renderResult.container.firstChild).toBeNull();
  });

  test("calls controller with clicked notification", () => {
    const viewModel = new BreakTimeNotificationOverviewViewModel();
    viewModel.showModal.Value = true;
    const controllerMock = mock<IBreakTimeNotificationOverviewController>();
    useBuilderMock([viewModel, controllerMock]);

    const renderResult = render(
      <Provider container={CoreDIContainer}>
        <BreakTimeNotificationOverview />
      </Provider>,
    );

    const notificationElement = renderResult.getByTestId(
      viewModel.shortBreakTimeNotifications[0].titleKey,
    );
    act(() => {
      fireEvent.click(notificationElement);
    });

    expect(controllerMock.selectNotification).toHaveBeenCalledTimes(1);
    expect(controllerMock.selectNotification).toHaveBeenCalledWith(
      viewModel.shortBreakTimeNotifications[0],
    );
  });
});
