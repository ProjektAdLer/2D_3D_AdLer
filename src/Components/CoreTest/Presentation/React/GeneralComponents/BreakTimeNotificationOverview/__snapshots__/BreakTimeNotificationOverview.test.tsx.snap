// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakTimeNotificationOverview should render 1`] = `
<div>
  <div
    class="z-50 flex items-center justify-center flex gap-2 m-2 w-fit h-fit text-adlertextgrey"
  >
    <div
      class="fixed top-0 bottom-0 left-0 right-0 flex items-center justify-center h-full bg-blacktrans"
    >
      <div
        class="z-50 p-2 rounded-lg bg-gradient-to-br from-adlerbggradientfrom to-adlerbggradientto max-w-[95%] max-h-[95%] overflow-hidden"
      >
        <div
          class="flex items-center justify-center w-full gap-2 p-1 pb-3 text-xl font-bold text-adlerdarkblue lg:roboto-black lg:text-2xl h-fit"
        >
          <div
            class="w-full"
          >
            overviewTitle
          </div>
          <button
            class="justify-center p-1 md:w-7 md:h-7 sm:w-6 sm:h-6 w-4 h-4 aspect-square"
          >
            <img
              alt="CloseButton"
              class="w-6 h-6"
              src="close.svg"
            />
          </button>
        </div>
        <div
          class="px-1 overflow-auto rounded-lg font-regular h-fit"
        >
          <div
            class=""
          >
            <div>
              <h1>
                shortBreaks
              </h1>
              <div
                class="grid grid-cols-5 gap-5"
              >
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="B3"
                    >
                      <img
                        alt="B3Thumbnail"
                        src="smallBreakContent-B3-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    B3
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="B7"
                    >
                      <img
                        alt="B7Thumbnail"
                        src="smallBreakContent-B7-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    B7
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="B8"
                    >
                      <img
                        alt="B8Thumbnail"
                        src="smallBreakContent-B8-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    B8
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="E2"
                    >
                      <img
                        alt="E2Thumbnail"
                        src="smallBreakContent-E2-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    E2
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="E4"
                    >
                      <img
                        alt="E4Thumbnail"
                        src="smallBreakContent-E4-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    E4
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="E10"
                    >
                      <img
                        alt="E10Thumbnail"
                        src="smallBreakContent-E10-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    E10
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="L1"
                    >
                      <img
                        alt="L1Thumbnail"
                        src="smallBreakContent-L1-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    L1
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="L3"
                    >
                      <img
                        alt="L3Thumbnail"
                        src="smallBreakContent-L3-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    L3
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="L5"
                    >
                      <img
                        alt="L5Thumbnail"
                        src="smallBreakContent-L5-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    L5
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="U3"
                    >
                      <img
                        alt="U3Thumbnail"
                        src="smallBreakContent-U3-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    U3
                  </div>
                </div>
              </div>
            </div>
            <div>
              <h1>
                mediumBreaks
              </h1>
              <div
                class="grid grid-cols-5 gap-5"
              >
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="S1"
                    >
                      <img
                        alt="S1Thumbnail"
                        src="mediumBreakContent-S1-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    S1
                  </div>
                </div>
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="S3"
                    >
                      <img
                        alt="S3Thumbnail"
                        src="mediumBreakContent-S3-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    S3
                  </div>
                </div>
              </div>
            </div>
            <div>
              <h1>
                longBreaks
              </h1>
              <div
                class="grid grid-cols-5 gap-5"
              >
                <div>
                  <div
                    class=""
                  >
                    <button
                      class="justify-center p-1 lg:w-16 lg:h-16 md:w-14 md:h-14 sm:w-10 sm:h-10 w-10 h-10 aspect-square flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-adlerdarkblue hover:bg-adleryellow lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 active:border-transparent text-adlerdarkblue font-regular border-b-2 border-r-2 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
                      data-testid="H1"
                    >
                      <img
                        alt="H1Thumbnail"
                        src="longBreakContent-H1-slide1.png"
                      />
                    </button>
                  </div>
                  <div>
                    H1
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
